modules/boltlib/lib/puppet/functions/run_task.rb

```diff
@@ -10,45 +10,38 @@
     type 'TargetOrTargets = Variant[String[1], Target, Array[TargetOrTargets]]'
   end
 
-  dispatch :run_task_type do
-    param 'Type[Task]', :task_type
+  dispatch :run_task do
+    param 'String[1]', :task_name
     param 'TargetOrTargets', :targets
     optional_param 'Hash[String[1], Any]', :task_args
   end
 
-  dispatch :run_named_task do
-    param 'String[1]', :task_type
+  # this is used from 'bolt task run'
+  dispatch :run_task_raw do
+    param 'String[1]', :task_name
     param 'TargetOrTargets', :targets
     optional_param 'Hash[String[1], Any]', :task_args
+    block_param
   end
```

Comment:
The function has these signatures now:
```ruby
dispatch :run_task do
  param 'String[1]', :task_name
  param 'TargetOrTargets', :targets
  optional_param 'Hash[String[1], Any]', :task_args
end

dispatch :run_task_raw do
  param 'String[1]', :task_name
  param 'TargetOrTargets', :targets
  optional_param 'Hash[String[1], Any]', :task_args
  block_param
end
```
note that they only differ in the `block_param` parameter - is this potentially ambiguous?